server/jetstream_cluster_test.go

```diff
@@ -9240,6 +9240,67 @@ func TestJetStreamClusterConsumerUpdates(t *testing.T) {
 	t.Run("Clustered", func(t *testing.T) { testConsumerUpdate(t, c.randomServer(), 2) })
 }
 
+func TestJetStreamClusterConsumerMaxDeliverUpdate(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "JSC", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{Name: "TEST", Subjects: []string{"foo"}, Replicas: 3})
+	require_NoError(t, err)
+
+	maxDeliver := 2
+	_, err = js.AddConsumer("TEST", &nats.ConsumerConfig{
+		Durable:       "ard",
+		AckPolicy:     nats.AckExplicitPolicy,
+		FilterSubject: "foo",
+		MaxDeliver:    maxDeliver,
+	})
+	require_NoError(t, err)
+
+	sub, err := js.PullSubscribe("foo", "ard")
+	require_NoError(t, err)
+
+	_, err = js.Publish("foo", []byte("Hello"))
+	require_NoError(t, err)
+
+	for i := 0; i <= maxDeliver; i++ {
```

Comment:
To reduce number of code lines, I would have made this an inner function:
```
checkMaxDeliver := func() {
    t.Helper()
    for i := 0; i <= maxDeliver; i++ {
        msgs, err := sub.Fetch(2, nats.MaxWait(10*time.Millisecond))
		if i < maxDeliver {
			require_NoError(t, err)
			require_Len(t, 1, len(msgs))
			_ = msgs[0].Nak()
		} else {
			require_Error(t, err, nats.ErrTimeout)
		}
	}
}
```
then call this here `checkMaxDeliver()`